### PR TITLE
Add a gflag for IO uring enable/disable

### PR DESCRIPTION
**Pull request by anand1976**
In case of IO uring bugs, we need to provide a way for users to turn it off.

Test plan:
Manually run db_bench with/without the option and verify the behavior

### CODE REVIEW

File: env/fs_posix.cc [hunk 1]
```diff
@@ -73,6 +76,11 @@
 #define EXT4_SUPER_MAGIC 0xEF53
 #endif
 
+#ifdef GFLAGS
+DEFINE_bool(rocksdb_io_uring_enable, true,
+            "If true, use IO uring if the platform supports it");
```
Comment (siying) [inline env/fs_posix.cc:81]: Can we explain that it is a backdoor temporary solution?

**Review by siying — APPROVED** [review]

**Comment by siying** [comment]
We need to validate if gflags is not configured or not available, it is still working. Not sure CI covers it.

**Comment by pdillinger** [comment]
> We need to validate if gflags is not configured or not available, it is still working. Not sure CI covers it.

build-linux-release-rtti checks that release build compiles without gflags 👍

**Comment by anand1976** [comment]
rocksdbjava build is faliing. We define -DGFLAGS, but don't link against libgflags I think

**Comment by pdillinger** [comment]
Would a weak symbol be better? For precedent see https://github.com/facebook/rocksdb/pull/7636/files and https://github.com/facebook/rocksdb/blob/main/port/jemalloc_helper.h

**Comment by anand1976** [comment]
> Would a weak symbol be better? For precedent see https://github.com/facebook/rocksdb/pull/7636/files and https://github.com/facebook/rocksdb/blob/main/port/jemalloc_helper.h

In this case, the weak symbol would be ```google::FlagRegisterer```, but we don't use that directly.

**Comment by pdillinger** [comment]
> > Would a weak symbol be better? For precedent see https://github.com/facebook/rocksdb/pull/7636/files and https://github.com/facebook/rocksdb/blob/main/port/jemalloc_helper.h
> 
> In this case, the weak symbol would be `google::FlagRegisterer`, but we don't use that directly.

I mean using a weak symbol to avoid gflags entirely

**Comment by pdillinger** [comment]
And Mark coded up a version based on making PosixFileSystem customizable, #8932

File: env/fs_posix.cc (hunk 1 above)
Comment (pdillinger) [inline env/fs_posix.cc:81]: Is this going to change the --help output of any tool using gflags that links with RocksDB? That seems intrusive

**Comment by anand1976** [comment]
@pdillinger A new version using your suggestion of a weak symbol. Haven't looked at Mark's PR yet, but that sounds like it could be a long term option.

**Review by pdillinger — APPROVED** [review]
👍

**Pull request closed without merging**